src/Symfony/Bundle/FrameworkBundle/Translation/Translator.php

```diff
@@ -54,8 +55,21 @@ class Translator extends BaseTranslator implements WarmableInterface
      *
      * @throws InvalidArgumentException
      */
-    public function __construct(ContainerInterface $container, MessageSelector $selector, $loaderIds = array(), array $options = array())
+    public function __construct(ContainerInterface $container, MessageSelector $selector, $defaultLocale = null, $loaderIds = array(), array $options = array())
     {
+        // BC 3.x, to be removed in 4.0 along with the $defaultLocale default value
+        if (is_array($defaultLocale) || 3 > func_num_args()) {
+            @trigger_error(sprintf('Method %s() takes the default locale as 3rd argument since version 3.3. Not passing it is deprecated and will trigger an error in 4.0.', __METHOD__), E_USER_DEPRECATED);
+            $options = $loaderIds;
+            $loaderIds = $defaultLocale;
+
+            if (!$container instanceof SymfonyContainerInterface) {
+                throw new \InvalidArgumentException(sprintf('A default locale must be passed as 3rd argument of %s() since version 3.3, the argument will be mandatory in 4.0.', __METHOD__));
```

Comment:
I'd suggest moving this before the trigger_error, and making the message simpler:
`Missing third $defaultLocale argument.`